mlflow/sklearn/__init__.py

```diff
@@ -567,7 +567,7 @@ def load_model(model_uri, dst_path=None):
     """
     local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
     flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
-    _add_code_from_conf_to_system_path(local_model_path, FLAVOR_NAME)
+    _add_code_from_conf_to_system_path(local_model_path, flavor_conf)
```

Comment:
Can we update `_add_code_from_conf_to_system_path` to validate the second argument (`conf`) is a dictionary?